### PR TITLE
Fix: update scripts to be compatible with dbt 0.17.1!

### LINKED ISSUE #25: `dbt-helper open` broke for dbt v0.15.0
I migrated to `dbt 0.15.0` and when i tried using `dbt-helper open <model_name` it's complaining that the `dbt.loader` module cannot be found.

I checked in the dbt code and indeed the module isn't here.

I don't know when/if I'll have time to look into this so I thought I'd pop an issue here so that it doesn't go missing.

There might be a case here to pin dbt requirements to a version < 15 at setup or so. If I can look into it I'll for sure contribute.

### PR DESCRIPTION
**Pull request by aescay**
## dbt-helper 0.4.0! 🎉 
Long live dbt-helper!!! This PR updates the scripts in dbt-helper to catch up with all the breaking changes introduced in newer dbt releases.

Bootstrapping dbt-helper from dbt core, makes it easy to leverage a good number of dbt's core functionality, but also makes it prone to breakage if it is not updated in a timely manner. I do still think this tight coupling is the ideal way to go because it would be incredibly difficult to replicate the same functionality from scratch.

## Change Overview
1. Most of the core scripts had to be updated to the renamed/changed methods of dbt core.
2. Logging was split from dbt core for overall simplicity and flexibility moving forward.
3. Renaming "archive" to "snapshot" to reflect the feature rename in dbt 0.14.0
4. Add new required default arg "--project-dir"
5. Adding `log_manager.reset_handlers()` as a required step before running `handle_and_check()`
6. Use relative path files instead of absolute paths for test `source-paths`
7. Update tests to reflect new path handling, using seed files, and updating the snapshot implementation

Resolves #25 (but should be in 0.17.1 to function properly, skipped over the versions in between)

### CODE REVIEW

File: test/integration/004_compare_snapshot_test/test_compare_snapshot.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+from test.integration.base import DBTIntegrationTest
+import os
+
+
+class CompareTest(DBTIntegrationTest):
```
Comment (mikekaminsky) [inline test/integration/004_compare_snapshot_test/test_compare_snapshot.py:5]: This seems much nicer now! :)

Comment (aescay) [reply test/integration/004_compare_snapshot_test/test_compare_snapshot.py:5]: I was introduced to dbt at the stage where `dbt archive` did not exist...so you can just imagine the look on my face when I saw `archive` as a subcommand and I was like "what is this sorcery?!?!" 🙃 Then I gladly found this page that I believe Drew may have left as a nice joke https://docs.getdbt.com/docs/building-a-dbt-project/archival/

Comment (mikekaminsky) [reply test/integration/004_compare_snapshot_test/test_compare_snapshot.py:5]: hahahahaha I love it

**Comment by aescay** [comment]
Can't seem to check the boxes on your review or edit the comment itself. Both edits were pushed!

**Comment by mikekaminsky** [comment]
Looks great! I'm going to merge and deploy and, if everything still works, it should end up on pypi shortly.

**Comment by bastienboutonnet** [comment]
Looks amazing! This is the PR I wanted to do but didn't finnd time to do and would have not done better!

**Pull request merged**